packages/yourloops/pages/profile/profile.tsx

```diff
@@ -0,0 +1,438 @@
+import apiClient from "../../lib/auth/api";
+import { getCurrentLocaleName, getLocaleShortname, availableLocales } from "../../lib/language";
+import { Profile, Roles, Settings, Units, User } from "../../models/shoreline";
+import { useAuth } from "../../lib/auth/hook/use-auth";
+import { Alert } from "@material-ui/lab";
+
+interface Errors {
+  firstName: boolean;
+  name: boolean;
+  mail: boolean;
+  password: boolean;
+  passwordConfirmation: boolean;
+  birthDate: boolean;
+}
+interface ApiReturnAlert {
+  message: string;
+  severity: "error" | "warning" | "info" | "success";
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    select: { padding: "1em 2em" },
+    button: { margin: "2em 1em" },
+    formControl: {
+      minWidth: 120,
+      textAlign: "right",
+    },
+    selectEmpty: {
+      marginTop: theme.spacing(2),
+    },
+    homeIcon: {
+      marginRight: "0.5em",
+    },
+    breadcrumbLink: {
+      display: "flex",
+    },
+    toolBar: {
+      display: "grid",
+      gridTemplateRows: "auto",
+      gridTemplateColumns: "auto auto auto",
+      paddingLeft: "6em",
+      paddingRight: "6em",
+    },
+    textField: {
+      "& input:disabled": {
+        backgroundColor: "#f7f7f8",
+      },
+    },
+    title: {
+      textAlign: "center",
+      color: theme.palette.primary.main,
+    },
+    inputTitle: { color: "black" },
+    inputProps: { textAlign: "right", padding: "1em 2em" },
+    container: {
+      border: "solid",
+      borderRadius: "15px",
+      marginTop: "32px",
+      // eslint-disable-next-line no-magic-numbers
+      borderColor: theme.palette.grey[300],
+      borderWidth: "1px",
+    },
+  })
+);
+
+const ProfileHeader = () => {
+  const { t } = useTranslation("yourloops");
+  const classes = useStyles();
+
+  return (
+    <Fragment>
+      <HeaderBar />
+      <AppBar position="static" color="secondary">
+        <Toolbar className={classes.toolBar}>
+          <Breadcrumbs aria-label={t("breadcrumb")}>
+            <Link className={classes.breadcrumbLink} color="textPrimary" href="/hcp">
+              <HomeIcon className={classes.homeIcon} />
+              {t("account-preferences")}
+            </Link>
+          </Breadcrumbs>
+        </Toolbar>
+      </AppBar>
+    </Fragment>
+  );
+};
+
+export const ProfilePage: FunctionComponent = () => {
+  const { t, i18n } = useTranslation("yourloops");
+  const classes = useStyles();
+  const history = useHistory();
+  const { user, setUser } = useAuth();
+
+  const [firstName, setFirstName] = useState<string>("");
+  const [name, setName] = useState<string>("");
+  const [mail, setMail] = useState<string>("");
+  const [locale, setLocale] = useState<string>(getCurrentLocaleName(i18n));
+  const [password, setPassword] = useState<string>("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState<string>("");
+  const [unit, setUnit] = useState<Units>(Units.mole);
```

Comment:
```suggestion
  const [unit, setUnit] = useState<Units>(Units.gram);
```
We will use mg/dL as default value for now.